include/nil/crypto3/math/domains/step_radix2_domain.hpp

```diff
@@ -47,29 +48,42 @@ namespace nil {
                 typedef typename FieldType::value_type field_value_type;
                 typedef ValueType value_type;
 
+                std::optional<
```

Comment:
Let's use unique_ptr<...> everywhere instead of std::optional. std::optional actually takes a bit more memory than unique_ptr.

Reply:
Moved all of them to `unique_ptr`.